### ISSUE TITLE
Does not authorize with pundit on batch actions (but does for member actions)

### ISSUE THREAD

**Issue by andrew-teeps** [activeadmin/activeadmin#4293]
Suppose I have a `Post`.

## This is what works
I can create a member action for post:

```ruby
member_action :approve, method: :put do
  resource.update(approved: true)
  redirect_to admin_post_path(resource), notice: 'Approved!'
end
```

and in my post policy define:

```ruby
def approve?
  user.admin?
end
```

And ActiveAdmin will correctly authorize the request.

## This doesn't seem to work

I can create a batch action for post:

```ruby
batch_action :approve_all do |ids|
  count = batch_action_collection.where(id: ids).update_all(approved: true)
  redirect_to admin_posts_path, notice: "Approved #{count} issues!"
end
```

And in my post policy define:

```ruby
def approve_all?
  user.admin?
end
```

But Active Admin is not authorizing the requests with pundit. Anyone can see and perform the batch action.

**Comment by timoschilling** [comment]
Batch actions don't support authorization at the moment!

Closed as double of #2856

**Comment by chofa17** [comment]
You can use a patch to solve this until we have a better solution!

```
batch_action :approve_all do |ids|
  authorize :approve_all, policy_class: PostPolicy

  count = batch_action_collection.where(id: ids).update_all(approved: true)
  redirect_to admin_posts_path, notice: "Approved #{count} issues!"
end
```

And in your Policy define it like it's shown below (record will contain the action to perform)
```
 def batch_action?
  return user.admin? if record == :approve_all

  false
end
```
